lib/lltz_michelson/lltz_michelson.ml

```diff
@@ -259,15 +259,75 @@ let rec compile : LLTZ.E.t -> Instruction.t =
          assert false
        | Fold_right { collection; init = Var acc, init_body; fold = Var var, fold_body }
          -> assert false
-       | Let_tuple_in { components; rhs; in_ } -> assert false
-       | Tuple row -> assert false
-       | Proj (tuple, path) -> assert false
-       | Update { tuple; component; update } -> assert false
+       | Let_tuple_in { components; rhs; in_ } ->
+         let rhs_instr = compile rhs in
+         let new_env = List.map components ~f:(fun (Var var) -> `Ident var) in
+         seq
+           [ rhs_instr
+           ; unpair_n (List.length components)
+           ; trace (Slot.let_all new_env ~in_:(compile in_))
+           ]
+       | Tuple row -> compile_row row
+       | Proj (tuple, path) ->
+         let lengths =
```

Comment:
I general try to fail asap, so we can forget about failure when reading the logic. Here:
```
let lengths =
  match tuple.desc, path with
   | LLTZ.E.Tuple row, Here list -> get_lengths row list
   | _ -> raise_s ...
   in
   ```